src/main/scala/beam/utils/beamToVia/IO/Writer.scala

```diff
@@ -0,0 +1,108 @@
+package beam.utils.beamToVia.IO
+
+import java.io.{File, PrintWriter}
+
+import beam.utils.beamToVia.viaEvent.ViaEvent
+
+import scala.collection.mutable
+
+object Writer {
+
+  def writeSeq[T](seq: Traversable[T], transform: T => String, outputPath: String): Unit = {
+    val pw = new PrintWriter(new File(outputPath))
+    seq.foreach(seqItem => pw.println(transform(seqItem)))
+    pw.close()
+  }
+
+  def writeViaEventsQueue[T](queue: mutable.PriorityQueue[T], transform: T => String, outputPath: String): Unit = {
+    val eventsCount = queue.size
+    Console.println(s"started writing $eventsCount events ...")
+    val progress = new ConsoleProgress("evenst written", eventsCount, 5)
+
+    val pw = new PrintWriter(new File(outputPath))
+    pw.println("<?xml version=\"1.0\" encoding=\"utf-8\"?>\n<events version=\"1.0\">")
+    while (queue.nonEmpty) {
+      progress.step()
+      val entry = queue.dequeue()
+      pw.println(transform(entry))
+    }
+    pw.println("</events>")
+    pw.close()
+
+    progress.finish()
+
+    Console.println("via events written into " + outputPath)
+  }
+
+  def writeSeqOfString(script: Traversable[String], outputPath: String): Unit = {
+    val pw = new PrintWriter(new File(outputPath))
```

Comment:
If exceptions happen the stream will remain opened. It's always good to make sure that resources are closed every time. You could use `FileUtils.using` for that purpose.

Reply:
ok, will use FileUtils.using